docs/contract-test-library.md

```diff
@@ -0,0 +1,541 @@
+  - [Tzip16OffChainContractTwo](#tzip16offchaincontracttwo)
+- **Wallet Contracts**
+  - [WalletContract](#walletcontract)
+  - [WalletAreYouThereContract](#walletareyouttherecontract)
+
+# Basic Contracts
+
+## IncrementContract
+
+[KT1NcdpzokZQY4sLmCBUwLnMHQCCQ6rRXYwS](https://better-call.dev/hangzhou2net/KT1NcdpzokZQY4sLmCBUwLnMHQCCQ6rRXYwS/code)
+
+This contract serves as the default introductory example on the [Ligo-WEB-IDE](https://ide.ligolang.org/p/CelcoaDRK5mLFDmr5rSWug)
+It has two endpoints, %decrement and %increment. The contract is used to demo addition and subtraction by a smart contract. This contract has neither an FA1.2 nor an FA2 interface.
+
+#### Entrypoints:
+  * decrement
+  * increment
+
+## MichelsonMapContract
+
+[KT1PAW3ghZyysrArcexyj6VUU7NZF8tHKoZR](https://better-call.dev/hangzhou2net/KT1PAW3ghZyysrArcexyj6VUU7NZF8tHKoZR/code)
+
+The contract supports a [Michelson Tutorial](https://tezostaquito.io/docs/michelsonmap). It has a default endpoint that takes a pair of an address and an amount of tez. 
+
+- [See the full tutorial](https://claudebarde.medium.com/?p=8d8be9930662)
+
+#### Entrypoints:
+  * default
+
+# Lambda Contracts
+We can send contract address, view method, and parameters as its own "view" to a simple lambda contract that always fails. We refer to this method as a "lambda view." The result of invoking our always-failing lambda contract is an error from the blockchain.
+
+That may not sound very useful, but the brilliant part is that the error we receive contains the information we requested! We can not incur a fee for requesting data or waiting for confirmation from the network to call view methods.
+
+Taquito internally contains a list of lambda contracts. Thus, there is no need to deploy a lambda contract if you are using Mainnet, Hangzhounet, or Granadanet. Taquito will detect the current network and use the appropriate lambda contract.
+
+## LambdaViewContract
+
+[KT1LNMrk8orMQ85zbwK25996dPhDxfSicvKh](https://better-call.dev/hangzhou2net/KT1LNMrk8orMQ85zbwK25996dPhDxfSicvKh/code)
+
+Not a supported FA1.2 contract. Almost an Fa2 interface but it is missing update_operators.
+
+#### Entrypoints:
+  * approve
+  * getAllowance
+  * getBalance
+  * getTotalSupply
+  * mint
+  * transfer
+
+## LambdaViewWithTokenContract
+
+[KT1Tf2JXZP8wXjdgndsgMKM1uW9M4CC5rbWL](https://better-call.dev/hangzhou2net/KT1Tf2JXZP8wXjdgndsgMKM1uW9M4CC5rbWL/code)
+
+This contact is another example of a Lambda contract, this time involving a token. It is not a supported FA1.2 contract but does have an FA2 interface.
+
+#### Entrypoints:
+  * balance_of
+  * token_metadata_registry
+  * transfer
+  * update_operators
+
+# Contracts with Maps
+
+## MapWithPairasMapContract
+
+[KT1BhFn1n1h4HJCxaRvoWEHPcp5UpAYbH3XN](https://better-call.dev/hangzhou2net/KT1BhFn1n1h4HJCxaRvoWEHPcp5UpAYbH3XN/code)
+
+A simple contract with a default entrypoint that takes unit. Not a supported FA1.2 contract. The storage looks like:
+
+```js
```

Comment:
The storage type in Michelson looks a bit different on BDC:
```
(pair
          (pair (address %theAddress)
                (map %theMap (pair nat address) (pair (mutez %amount) (int %quantity))))
          (int %theNumber));
```

Reply:
i took the storage from the display in BCD for "storage" on the, for example, https://better-call.dev/hangzhou2net/KT1BhFn1n1h4HJCxaRvoWEHPcp5UpAYbH3XN/storage page.

which form of presenting the storage is more useful?

Reply:
I have redid the storage presentations to use the Michelson format.